replica/replica.go

```diff
@@ -1413,3 +1401,12 @@ func (r *Replica) SetReplicaMode(mode string) error {
 	}
 	return nil
 }
+
+// SetCheckpoint ...
+func (r *Replica) SetCheckpoint(snapshotName string) error {
+	r.Lock()
+	defer r.Unlock()
+	r.info.Checkpoint = snapshotName
```

Comment:
don't we need to check that snapshotName is part of the chain?

Reply:
This check is performed at the controller in GetLastSnapshot(). But it can be checked at replica also.